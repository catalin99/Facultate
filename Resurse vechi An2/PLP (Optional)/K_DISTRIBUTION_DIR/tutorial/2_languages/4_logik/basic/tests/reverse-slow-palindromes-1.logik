append(nil,L,L).
append(cons(H,T),L,cons(H,Z)) :- append(T,L,Z).

reverse(nil,nil).
reverse(cons(H,T),L) :-
  reverse(T,T1),
  append(T1,cons(H,nil),L).

// Next should generate all palindromes.  However, since there are infinitely
// many of them, use a bound, say 5, to only see a few.
?- reverse(L,L).
