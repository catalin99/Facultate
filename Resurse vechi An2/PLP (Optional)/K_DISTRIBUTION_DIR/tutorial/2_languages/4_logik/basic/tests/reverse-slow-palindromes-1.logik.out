Search results:

Solution 1:
Solution:Map -->
L:Term |-> ( cons ( V0 , ( cons ( V0 , ( nil , .Terms ) ) , .Terms ) ) )

Solution 2:
Solution:Map -->
L:Term |-> ( cons ( V0 , ( cons ( V1 , ( cons ( V0 , ( nil , .Terms ) ) , .Terms ) ) , .Terms ) ) )

Solution 3:
Solution:Map -->
L:Term |-> ( cons ( V0 , ( cons ( V1 , ( cons ( V1 , ( cons ( V0 , ( nil , .Terms ) ) , .Terms ) ) , .Terms ) ) , .Terms ) ) )

Solution 4:
Solution:Map -->
L:Term |-> ( cons ( V0 , ( nil , .Terms ) ) )

Solution 5:
Solution:Map -->
L:Term |-> nil

